### PR TITLE
update bootstrap v4.1 generator template

### LINKED ISSUE #1561: Remove initializers from Bootstrap and Foundation
We don't have plans to support these initializers anymore. We have issues that are not related to Simple Form like bugs related to styles and we must update the initializer on each updates of these Frameworks.

For me, it makes sense to use Bootstrap, Foundation and others as a reference to make Simple Form flexible (see https://github.com/plataformatec/simple_form/pull/1337#issuecomment-350889176), as we are doing with Bootstrap 4, for instance https://github.com/plataformatec/simple_form/pull/1552 and https://github.com/plataformatec/simple_form/pull/1559.

What do you folks think?

### PR DESCRIPTION
**Pull request by m5o**
simple form now 💜 bootstrap 4.1

:tv: [Demo preview](https://simple-form-bootstrap4.herokuapp.com)

The bootstrap generator is extracted from:
https://github.com/rafaelfranca/simple_form-bootstrap/pull/56

* supported form wrapper all covered by *tests*
  * vertical form
  * horizontal form
  * inline form
  * custom fields form
  * input groups form
    * requires [custom input group component](https://github.com/rafaelfranca/simple_form-bootstrap/pull/56/files#diff-76a3f9bcc0567de956d6a9d798f1de7d)
  * floating labels form

:link: https://github.com/rafaelfranca/simple_form-bootstrap/pull/56
* Resolves #1561
* Resolves #1337 

---
📌 Bootstrap generator template is maintained by the community around [simple_form-bootstrap](https://github.com/rafaelfranca/simple_form-bootstrap)
⛏ All future development, tests, and organization should happen there.

### CODE REVIEW

File: lib/generators/simple_form/templates/config/initializers/simple_form_bootstrap.rb [hunk 1]
```diff
@@ -1,166 +1,439 @@
...
-  config.wrappers :vertical_form, tag: 'div', class: 'form-group', error_class: 'has-error', valid_class: 'has-success' do |b|
+  # Method used to tidy up errors. Specify any Rails Array method.
+  # :first lists the first message for each field.
+  # :to_sentence to list all errors for each field.
+  config.error_method = :to_sentence
+
+  # add validation classes to `input_field`
+  config.input_field_error_class = 'is-invalid'
+  config.input_field_valid_class = 'is-valid'
+
```
Comment (feliperenan) [inline lib/generators/simple_form/templates/config/initializers/simple_form_bootstrap.rb]: Duplicate blank line ✂️

Comment (feliperenan) [reply lib/generators/simple_form/templates/config/initializers/simple_form_bootstrap.rb]: I see it in other places, so I think it was intentional. Just one line for me it's ok, feel free to make this change 😄

Comment (m5o) [reply lib/generators/simple_form/templates/config/initializers/simple_form_bootstrap.rb]: I've added two blank lines before each form block (vertical, horizontal, inline, custom, group and floating)

File: lib/generators/simple_form/templates/config/initializers/simple_form_bootstrap.rb [hunk 2]
```diff
@@ -1,166 +1,439 @@
...
-# more about custom components.
+# See https://github.com/plataformatec/simple_form#custom-components
+# to know more about custom components.
 # Dir[Rails.root.join('lib/components/**/*.rb')].each { |f| require f }
-#
+
 # Use this setup block to configure all options available in SimpleForm.
 SimpleForm.setup do |config|
+  # Default class for buttons
+  config.button_class = 'btn'
```
Comment (jonathanhefner) [inline lib/generators/simple_form/templates/config/initializers/simple_form_bootstrap.rb]: Should this be `'btn btn-primary'` or `'btn btn-default'`?

Comment (m5o) [reply lib/generators/simple_form/templates/config/initializers/simple_form_bootstrap.rb]: Answered here: https://github.com/rafaelfranca/simple_form-bootstrap/pull/56#discussion_r183664638

**Pull request merged**